### PR TITLE
print timings on ctrl+c exit

### PR DESCRIPTION
**Pull request by wbpxre150**
Currently the timing stats do not print when you use CTRL+C to exit interactive mode. This is the simplest way to fix that.

### CODE REVIEW

File: examples/main/main.cpp [hunk 1]
```diff
@@ -36,6 +37,8 @@ void sigint_handler(int signo) {
         if (!is_interacting) {
             is_interacting=true;
         } else {
+            llama_print_timings(ctx);
+            llama_free(ctx);
```
Comment (DannyDaemonic) [inline examples/main/main.cpp:41]: There's no harm in it, but I probably wouldn't bother to free `ctx` right before an `_exit` in an sigint handler. `_exit`'s purpose is to terminate the program immediately. Typically, you either clean up your memory properly and `exit`, or you don't and you use `_exit`.

Comment (wbpxre150) [reply examples/main/main.cpp]: No worries, I wasn't aware of this. Its removed. :)

File: examples/main/main.cpp [hunk 2]
```diff
@@ -36,6 +37,7 @@ void sigint_handler(int signo) {
         if (!is_interacting) {
             is_interacting=true;
         } else {
+            llama_print_timings(ctx);
```
Comment (ggerganov) [inline examples/main/main.cpp:40]:
```suggestion
            llama_print_timings(*g_ctx);
```

File: examples/main/main.cpp [hunk 3]
```diff
@@ -25,6 +25,7 @@
 #endif
 
 static console_state con_st;
+llama_context * ctx;
```
Comment (ggerganov) [inline examples/main/main.cpp:28]:
```suggestion
static llama_context ** g_ctx;
```

File: examples/main/main.cpp [hunk 4]
```diff
@@ -93,8 +95,6 @@ int main(int argc, char ** argv) {
 //    params.prompt = R"(// this function checks if the number n is prime
 //bool is_prime(int n) {)";
 
-    llama_context * ctx;
-
     // load the model
```
Comment (ggerganov) [inline examples/main/main.cpp:98]:
```suggestion
    llama_context * ctx;
    g_ctx = &ctx;

    // load the model
```

**Review by ggerganov — CHANGES_REQUESTED** [review]
My suggestions are pretty minor, but I just want to keep `ctx` a local variable.
So I add a global pointer to observe it

**Comment by wbpxre150** [comment]
@ggerganov 
I have committed these changes, all works as expected, thanks for the suggestion. Don't know why I didn't think of this myself. :)

**Review by ggerganov — APPROVED** [review]

**Pull request merged**